### PR TITLE
Molar Mass: Resolve conflicting triggers issue and add new triggers

### LINKED ISSUE #4240: Molar Mass: Conflicting with Periodic table IA (Suggestion)
**Issue:**
We already have a related IA [periodic table](https://duck.co/ia/view/periodic_table) and some of the triggers of this IA conflicts with that IA 

For ex: Screenshots:
screenshot from 2017-06-02 17-45-03 

screenshot from 2017-06-02 17-46-22

~~There is **one more slight issue** this IA has: It doesn't display the tab on the top.~~ (**Fixed**)

~~imageedit_1_7716006707~~

People to notify:
@joshliu @pjhampton @moollaza 

**Suggestion**: Reduce some of the triggers like `atomic mass` and `atomic weight`. Let this IA trigger only when user types `molar mass <element|compound>` so it doesn't conflict with the periodic table IA.

**P.S.** With all due respect to the hard work done by @joshliu this IA can be merged with the periodic table IA or vice versa at any later stage.

------
IA Page:  http://duck.co/ia/view/molar_mass 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @joshliu

### PR DESCRIPTION
**Pull request by joshliu**
## Description of changes
Removed "atomic mass of" and "atomic weight of" triggers that were conflicting with Periodic Table IA, and added triggers to handle queries such as "H2SO4 molar mass" and "whats the molar mass of water?".

## Related Issues and Discussions

Fixes #4240

## People to notify

@pjhampton @adityatandon007 @moollaza 

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/molar_mass

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Goodie/MolarMass.pm [hunk 1]
```diff
@@ -7,20 +7,25 @@ use warnings;
...
-triggers start => 'molar mass of', 'atomic mass of', 'atomic weight of';
+triggers any => 'molar mass of';
+triggers end => 'molar mass';
 
 # Handle statement
 handle remainder => sub {
 
     my $remainder = $_;
 
+    $remainder =~ s/(|what is|whats|what\'s|the|of|for|\?|)//g;
```
Comment (moollaza) [inline lib/DDG/Goodie/MolarMass.pm]: Remove the leading/trailing `|`: `s/(what is|what\'?s|the|of|for|\?)//g;`

**Comment by moollaza** [comment]
Deploying to Beta. I think that regex needs to be corrected...

**Comment by adityatandon007** [comment]
@joshliu great work! :100:
Everything looks fine except one thing, it doesn't support queries like `molar mass nacl`

screenshot from 2017-06-15 17-45-51

My first instinct was that it should have triggered because I remembered you were using `triggers any` but when I looked and checked it, I found the minor glitch why it isn't triggering, so you are using `triggers any => molar mass of` and `triggers end => molar mass` and then you have used regex to expand the query base. I think it would have been better if you have removed that `triggers end` and only used `triggers any => molar mass` because your regex already takes care of that `of` word and plus you could also support queries like `molar mass nacl`. 
I haven't tried that, since perl is still very new to me and I think @moollaza might guide you better.

**Comment by moollaza** [comment]
@adityatandon007 to handle that query we can change `triggers end => molar mass` to `triggers startend => molar mass` 👍

**Pull request merged**